src/wallet/wallet.ts

```diff
@@ -157,6 +206,66 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     return this.authToken;
   }
 
+  /**
+   * Returns the balance for each token in tx, if the input/output belongs to this wallet
+   *
+   * This method is meant to keep compatibility with the old facade
+   *
+   * @param {Object} tx Transaction data with array of inputs and outputs
+   *
+   * @return {Object} Object with each token and it's balance in this tx for this wallet
+   **/
+  async getTxBalance(tx: WsTransaction, optionsParam = {}): Promise<{[tokenId: string]: number}> {
+    const options = Object.assign({ includeAuthorities: false }, optionsParam);
+
+    const addressList: GetAddressesObject[] = [];
+
+    for await (let address of this.getAllAddresses()) {
+      addressList.push(address);
+    }
+
+    const addresses: string[] = addressList.map((address: GetAddressesObject) => address.address);
+
+    const balance = {};
+    for (const txout of tx.outputs) {
+      if (wallet.isAuthorityOutput(txout)) {
+        if (options.includeAuthorities) {
+          if (!balance[txout.token]) {
+            balance[txout.token] = 0;
+          }
+        }
+        continue;
+      }
+      if (txout.decoded && txout.decoded.address
+          && addresses.includes(txout.decoded.address)) {
+        if (!balance[txout.token]) {
+          balance[txout.token] = 0;
+        }
+        balance[txout.token] += txout.value;
+      }
+    }
+
+    for (const txin of tx.inputs) {
+      if (wallet.isAuthorityOutput(txin)) {
+        if (options.includeAuthorities) {
+          if (!balance[txin.token]) {
+            balance[txin.token] = 0;
+          }
+        }
+        continue;
+      }
+      if (txin.decoded && txin.decoded.address
+          && addresses.includes(txin.decoded.address)) {
+        if (!balance[txin.token]) {
+          balance[txin.token] = 0;
+        }
+        balance[txin.token] -= txin.value;
```

Comment:
Should we have tests for these new methods?

Reply:
Added tests for the `getTxBalance` method in fc56280984abb512c6e7443a383d9e92c98a8396